com.ibm.streamsx.topology/opt/python/packages/streamsx/topology/topology.py

```diff
@@ -83,6 +84,51 @@ def isolate(self):
         op.addInputPort(outputPort=self.oport)
         oport = op.addOutputPort()
         return Stream(self.topology, oport)
+    
+    def parallel(self, width):
+        """
+        Marks operator as parallel output with width
+        :param: width
+        :returns: Stream
+        """
+        iop = self.isolate()
+               
+        op2 = self.topology.graph.addOperator("$Parallel$")
+        op2.addInputPort(outputPort=iop.getOport())
+        oport = op2.addOutputPort(width)
+        return Stream(self.topology, oport)
+
+    def endParallel(self):
+        """
+        Marks end of operators as parallel output with width
+        :returns: Stream
+        """
+        op = self.topology.graph.addOperator("$EndParallel$")
```

Comment:
In the JAA, the endParallel code is as follows:
``` Java
    @Override
    public TStream<T> endParallel() {
        BOutput end = output();
        if(end instanceof BUnionOutput){
            end = builder().addPassThroughOperator(end);
        }
        end = builder().isolate(end);
        return addMatchingStream(builder().unparallel(end));
    }
```
There's a check which ensures that a $Union$ never immediately precedes an $EndParallel$ since it breaks the code generator. You should do something similar. If you see a $Union$ right before an $EndParallel$, add a "no-op" pass through operator as a buffer.